src/Application/Channels/Commands/CreateChannelCommand.cs

```diff
@@ -89,10 +81,37 @@ public async Task<Guid> Handle(CreateChannelCommand request, CancellationToken c
             entity.RangeRule = "*";
         }
 
+        entity.ActiveRevision = await GetActiveRevision(request, entity, cancellationToken);
+
         _context.Channels.Add(entity);
 
         await _context.SaveChangesAsync(cancellationToken);
 
         return entity.Id;
     }
+
+    private async Task<Revision?> GetActiveRevision(CreateChannelCommand request, Channel entity, CancellationToken cancellationToken)
+    {
+        Revision? revision = null;
+
+        if (entity.RevisionSelectionStrategy == ChannelRevisionSelectionStrategy.UseRangeRule &&
+                    request.ActiveRevisionId is null)
+        {
+            var revisions = _context.Revisions
+                .Where(c => c.AppId == request.AppId).ToList();
+
+            revision = RevisionRangeRule.Parse(entity.RangeRule).Match(revisions);
+        }
+
+        if (entity.RevisionSelectionStrategy == ChannelRevisionSelectionStrategy.UseSpecifiedRevision &&
```

Comment:
same questions as above what happens if the UseSpecifiedRevision is true but there is no activeRevisionId or what if UseSpecifiedRevision is true and UseRangeRule is also true